### PR TITLE
DRILL-6760: Retain original exception in Verbose Error Message

### PR DESCRIPTION
**Pull request by lushuifeng**
please see [DRILL-6760](https://issues.apache.org/jira/browse/DRILL-6760)
There are some cases that verbose error message in profile page contains no original exception stack trace.

### CODE REVIEW

**Comment by lushuifeng** [comment]
@arina-ielchiieva Could you please review this?
The tests are succeeded but timeout

File: common/src/main/java/org/apache/drill/common/util/DrillExceptionUtils.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+ */
+package org.apache.drill.common.util;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.drill.exec.proto.UserBitShared;
+
+import java.lang.reflect.Constructor;
+import java.util.Arrays;
+
+public class DrillExceptionUtils {
```
Comment (arina-ielchiieva) [inline common/src/main/java/org/apache/drill/common/util/DrillExceptionUtils.java]: 1. Please add java doc for the whole util class.
2. Please add unit tests for each public static method.
3. DrillExceptionUtil or DrillExceotionUtils?
4. Maybe move this helper class to drill-commons module, package - `org.apache.drill.common.util` ?

File: common/src/main/java/org/apache/drill/common/util/DrillExceptionUtils.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      }
+      throwable.setStackTrace(stackTrace);
+      return throwable;
+    } catch (Throwable t) {
+      logger.error("Failed to get throwable from PB", t);
+      return null;
+    }
+  }
+
+  private static Throwable getInstance(String className, String message, Throwable inner) throws ReflectiveOperationException {
```
Comment (arina-ielchiieva) [inline common/src/main/java/org/apache/drill/common/util/DrillExceptionUtils.java]: Please add java doc.

File: exec/java-exec/src/test/java/org/apache/drill/exec/ProfileTest.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  public static void setup() throws Exception {
+    startCluster(ClusterFixture.builder(dirTestWatcher));
+  }
+
+  @Test
+  public void verboseErrorTest() throws Exception {
+    DrillbitContext context = cluster.drillbit().getContext();
+    PersistentStore<UserBitShared.QueryProfile> completed = context.getProfileStoreContext().getCompletedProfileStore();
+    String sql = "select * from cp.`jsoninput/DRILL-2350.json`";
+    QueryBuilder.QuerySummaryFuture future = client.queryBuilder().sql(sql).futureSummary();
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/test/java/org/apache/drill/exec/ProfileTest.java]: Why are getting future summary? Why not just get `QueryBuilder.QuerySummary testQuerySummary = queryBuilder().sql(testQuery).run();`?

Also you can add check that `assertTrue(testQuerySummary.failed());`

File: exec/java-exec/src/test/java/org/apache/drill/exec/ProfileTest.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    QueryBuilder.QuerySummary summary = future.get();
+    String queryId = summary.queryIdString();
+    UserBitShared.QueryProfile profile = completed.get(queryId);
+    if (profile == null) {
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/test/java/org/apache/drill/exec/ProfileTest.java]: How do you know that profile is absent sometimes if this does not fail on your laptop? Not sure if we need test that fails randomly and adding logger looks more like a hack. If we cannot stabilize the test, then we might consider just removing it.

File: exec/java-exec/src/test/java/org/apache/drill/exec/ProfileTest.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+      logger.error("No profile found, queryId:" + queryId);
+      return;
+    }
+    String verboseError = profile.getVerboseError();
+    String extraError = verboseError.substring(profile.getError().length());
+    Pattern pattern = Pattern.compile("[a-z0-9]+(\\.[a-zA-Z0-9$]+)+\\(\\):\\d+");
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/test/java/org/apache/drill/exec/ProfileTest.java]: Please add comment explaining the pattern

File: exec/java-exec/src/test/java/org/apache/drill/exec/ProfileTest.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.junit.Assert.fail;
+
+public class ProfileTest extends ClusterTest {
+  static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(ProfileTest.class);
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/test/java/org/apache/drill/exec/ProfileTest.java]: private? or actually remote the logger at all.

File: common/src/main/java/org/apache/drill/common/util/DrillExceptionUtils.java [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+public class DrillExceptionUtils {
+  static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(DrillExceptionUtils.class);
+
+  public static Throwable getThrowable(UserBitShared.DrillPBError error) {
+    UserBitShared.ExceptionWrapper wrapper = error.getException();
+    return getThrowable(wrapper);
+  }
+
+  /**
+   * recreate throwable from exception PB which received from remote or local node.
```
Comment (arina-ielchiieva) [inline common/src/main/java/org/apache/drill/common/util/DrillExceptionUtils.java]: recreate -> Recreate
PB -> protobuf

**Comment by arina-ielchiieva** [comment]
@lushuifeng do you have Apache Jira account? I want to assign Jira (https://issues.apache.org/jira/browse/DRILL-6760) to you.

File: exec/java-exec/src/test/java/org/apache/drill/exec/ProfileTest.java (hunk 4 above)
Comment (lushuifeng) [reply exec/java-exec/src/test/java/org/apache/drill/exec/ProfileTest.java]: The test always failed and verbose error is absent, What's interesting is that the queryId sometimes is not stored in Completed Store in Travis tests which is not reproducible on my laptop.
You can launch the sqlline and run the same sql test in `ProfileTest` and take a look at the error page.
I will provide more details tomorrow.

Comment (arina-ielchiieva) [reply exec/java-exec/src/test/java/org/apache/drill/exec/ProfileTest.java]: Then just remove the unit test.

**Comment by lushuifeng** [comment]
@arina-ielchiieva Changes are made according to your suggestions, could you please take another look? (Building job timeouts)

Here is the comparison of before and after the change with the same query.
screen shot 2018-11-08 at 11 19 29

**Comment by arina-ielchiieva** [comment]
+1, please squash the commits.

**Comment by lushuifeng** [comment]
Done.

**Pull request closed without merging**